mainProject/controllers/users.ts

```diff
@@ -1,119 +1,76 @@
-        isDeleted: false
-    }
-];
-
-const getAutoSuggestUsers = (
-    loginSubstring: string,
-    limit: number
-): Array<User> | [] => {
-    const availableRecords = localStorage
-        .filter((obj) => !obj.isDeleted)
-        .sort((firstObj, secondObj) => {
-            if (firstObj.login > secondObj.login) return 1;
-            if (firstObj.login < secondObj.login) return -1;
-            return 0;
-        });
-    const result = [];
-    for (let i = 0; i < availableRecords.length; i++) {
-        if (result.length === limit) {
-            break;
-        }
-        if (availableRecords[i].login.includes(loginSubstring)) {
-            result.push(availableRecords[i]);
-        }
-    }
-
-    return result;
-};
-
-const getUsers = (req, res) => {
-    const query = req.query;
-    if (
-        !query.limit ||
+const getUsers = async (req, res) => {
+  const query = req.query;
+  if (
+    !query.limit ||
     !Number.isInteger(Number(query.limit)) ||
     Number(query.limit) < 0
-    ) {
-        return res.status(400).send('Property limit should be Integer and not less than 0.');
-    }
-    if (!query.login) {
-        return res.status(400).send('Property login should be in the query and have a value.');
-    }
-    const responseData = getAutoSuggestUsers(
-        String(query.login),
-        Number(query.limit)
-    );
-    res.json(responseData);
+  ) {
+    return res
+      .status(400)
+      .send("Property limit should be Integer and not less than 0.");
+  }
+  if (!query.login) {
+    return res
+      .status(400)
+      .send("Property login should be in the query and have a value.");
+  }
+  const responseData = await userService.getAutoSuggestUsers(
+    String(query.login),
+    Number(query.limit)
+  );
+  res.json(responseData);
 };
 
-const createUser = (req, res) => {
-    const user = req.body;
-    const dataKeys = Object.keys(user);
-    if (!dataKeys.length || isAvailableProperty(dataKeys)) {
-        return res.status(400).send('Incorrect request data.');
-    }
-    const userId = v4();
-    const userWithId: User = {
-        id: userId,
-        ...user,
-        isDeleted: false
-    };
-    localStorage.push(userWithId);
+const createUser = async (req, res) => {
+  const user = req.body;
 
-    res.send(`User with login ${userWithId.login} was added to the Database.`);
+  const data = await userService.createUser(user);
+
+  res.send(`User with login ${data.login} was added to the Database.`);
 };
 
-const getUserById = (req, res) => {
-    const { id } = req.params;
-    const user = localStorage.find((obj) => obj.id === id && !obj.isDeleted);
-    if (!user) {
-        return res.send('There is no record in the Database with this ID.');
-    }
-    res.json(user);
+const getUserById = async (req, res) => {
+  const { id } = req.params;
+  const user = await userService.getUserById(id);
+
+  if (!user) {
+    return res.send("There is no record in the Database with this ID.");
```

Comment:
That will be counted as response with status code 200, but with different type (string). I would say that could be some issues for users during using it. smth like stus 404 and `{message: 'There is no record in the Database with this ID'}`

Reply:
Or also could be useful to create some common structure of response. Smth like
```
{
  success: true/false,
  message: 'Blah blah',
  data: [...]
}
```

Reply:
This was intentional, since when requesting we specify the id in the correct form and the server is simply not able to find a user with a similar id, but after the proposal to use the 404 status code, I see that it is worth using the 404 status code instead of 200

Reply:
Yep, Also I would suggest to read about response formats. Because for example If I will use api with such kind of responses it will be not really easy to process such responses. 
For example I should now about all possible response messages that could be sent from api. You can find examples of such formats here https://stackoverflow.com/questions/12806386/standard-json-api-response-format#:~:text=JSON%2DRPC%202.0%20defines%20a,after%20working%20with%20REST%20APIs.&text=The%20basic%20framework%20suggested%20looks,as%20defined%20is%20too%20limited.&text=This%20format%20is%20consistent%20with%20the%20JSON%2DRPC%20specification.